aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,764 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import io
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import dask.array as da
+import numpy as np
+from dask import delayed
+from lxml.etree import Element
+from readlif import utilities
+from readlif.reader import LifFile
+
+from .reader import Reader
+from .. import exceptions, types
+from ..buffer_reader import BufferReader
+from ..constants import Dimensions
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class LifReader(Reader):
+    """
+    LifReader wraps readlif.reader to provide the same reading capabilities but abstracts the specifics of using the
+    backend library to create a unified interface. This enables higher level functions to duck type the File Readers.
+
+    Parameters
+    ----------
+    data: types.FileLike
+        A string or path to the LIF file to be read.
+    chunk_by_dims: List[str]
+        The dimensions to use as the for mapping the chunks / blocks.
+        Default: [Dimensions.SpatialZ, Dimensions.SpatialY, Dimensions.SpatialX]
+        Note: SpatialY and SpatialX will always be added to the list if not present.
+    S: int
+        If the image has different dimensions on any scene from another, the dask array construction will fail.
+        In that case, use this parameter to specify a specific scene to construct a dask array for.
+        Default: 0 (select the first scene)
+    """
+
+    LIF_MAGIC_BYTE = 0x70
+    LIF_MEMORY_BYTE = 0x2a
+
+    ########################################################
+    #
+    #  Note LifFile treats scenes as separate images in the lif file.
+    #  So once a Scene/Image is loaded the image data is retrieved
+    #  2D XY plane by 2D XY plane.
+    #
+    ########################################################
+
+    @property
+    def dims(self) -> str:
+        """
+        The dimensions for a lif file.
+
+        Returns
+        -------
+        str
+            "STCZYX"
+        """
+        return Dimensions.DefaultOrder  # forcing 6 D
+
+    @staticmethod
+    def _compute_offsets(lif: LifFile) -> Tuple[List[np.ndarray], np.ndarray]:
+        """
+        Compute the offsets for each of the XY planes so that the LifFile object doesn't need
+        to be created for each XY image read.
```

Comment:
I think you took my last comment of:

> nit: "YX image read"

and changed every `YX` to `XY`

That is not what I was saying, I meant, it should be `plane` not `image`. Should have been more clear.

I assume this format stores data in `YX` and not `XY` and if so please change back all `XY` to `YX`.

Reply:
Thanks for clarifying, done.